components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -142,6 +145,22 @@ handler.on(PanelActions.cancelConnectToSite.getType(), async (store: Store, payl
   apiProxy.closeUI()
 })
 
+handler.on(PanelActions.approveHardwareTransaction.getType(), async (store: Store, txInfo: TransactionInfo) => {
+  const hardwareAccount = await findHardwareAccountInfo(txInfo.fromAddress)
+  if (!hardwareAccount || !hardwareAccount.hardware) {
+    return
+  }
+  const apiProxy = await getAPIProxy()
+  apiProxy.closePanelOnDeactivate(false)
+  if (hardwareAccount.hardware.vendor === LEDGER_HARDWARE_VENDOR) {
+    await signLedgerTransaction(apiProxy, hardwareAccount.hardware.path, txInfo)
+  } else if (hardwareAccount.hardware.vendor === TREZOR_HARDWARE_VENDOR) {
+    await signTrezorTransaction(apiProxy, hardwareAccount.hardware.path, txInfo)
+  }
+  apiProxy.closePanelOnDeactivate(true)
+  await refreshWalletInfo(store)
```

Comment:
is there a lighter amount of actions and data we could retrieve and dispatch, or do we really need to refresh everything? 

`refreshWalletInfo` has issues in that it updates a lot of data (balances, prices, price history, site permissions, metamask installed, default wallet status) and also dispatches it individually, causing perf issues.

Perhaps you only need `refreshTransactionHistory` and maybe `refreshBalancesAndPrices`?

Reply:
done